k8sdeps/configmapandsecret/configmapfactory.go

```diff
@@ -142,10 +144,22 @@ func addKvToConfigMap(configMap *v1.ConfigMap, keyName, data string) error {
 	if errs := validation.IsConfigMapKey(keyName); len(errs) != 0 {
 		return fmt.Errorf("%q is not a valid key name for a ConfigMap: %s", keyName, strings.Join(errs, ";"))
 	}
-	if _, entryExists := configMap.Data[keyName]; entryExists {
-		return fmt.Errorf("cannot add key %s, another key by that name already exists: %v", keyName, configMap.Data)
+
+	keyExistsErrorMsg := "cannot add key %s, another key by that name already exists: %v"
+
+	if utf8.Valid([]byte(data)) {
```

Comment:
Can you add some explanation of using `utf8.Valid` to check if it is binary or not?

Reply:
I added some comments. Let me know if that is not sufficient. Also see

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/generate/versioned/configmap.go#L262 and for reference